cmd/util/api.go

```diff
@@ -79,26 +67,68 @@ func GetAPIClientV2(cmd *cobra.Command, cfg types.Bacalhau) (clientv2.API, error
 		clientv2.WithHeaders(headers),
 	}
 
-	authTokenPath, err := cfg.AuthTokensPath()
+	var resolvedAuthToken *apimodels.HTTPCredential
+
+	// Check if the credentials are valid
+	apiKeyOrBasicAuthFlowEnabled, credentialScheme, credentialString, err := resolveAuthCredentials(
+		apiAuthAPIKey,
+		basicAuthUsername,
+		basicAuthPassword,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("authentication error: %v", err)
+	}
+
+	legacyAuthTokenFilePath, err := cfg.AuthTokensPath()
```

Comment:
am I reading this correctly that the legacy auth logic has higher precedence than sso tokens?

Reply:
No, SSO tokens have higher precedence compared to legacy tokens.